src/main/scala/pipedsl/Parser.scala

```diff
@@ -32,26 +70,31 @@ class Parser extends RegexParsers with PackratParsers {
     "\"" ~> "[^\"]*".r <~ "\"" ^^ {n => EString(n)}
 
   private def toInt(n: Int, base: Int, bits: Option[Int], isUnsigned: Boolean): EInt = {
-    val e = EInt(n, base, if (bits.isDefined) bits.get else log2(n))
-    e.typ = Some(TSizedInt(e.bits, unsigned = isUnsigned))
+    val e = EInt(n, base, if (bits.isDefined) bits.get else  -1)
+    e.typ = bits match {
+      case Some(b) => Some(TSizedInt(TBitWidthLen(b), SignFactory.ofBool(!isUnsigned)))
+      case None if isUnsigned => Some(TSizedInt(genBitVar(), TUnsigned()))
```

Comment:
Why is there no case for `!isUnsigned`  when `bits` is None?

Reply:
you should be able to mark a constant as unsigned, but still use inference to determine the width. I've added a comment in the code explaining this.